cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewRunner.java

```diff
@@ -69,8 +71,16 @@
   List<MetricTimeSeries> getMetrics();
 
   /**
-   * Get the logs for the preview run represented by this {@link PreviewRunner}.
-   * @return the logs
+   * Get the program run id of the preview run
+   * @return the {@link ProgramRunId} associated with the preview
    */
-  List<LogEntry> getLogs();
+  @Nullable
```

Comment:
When `DEPLOY_FAILED` occurs we should not call logs endpoint right? PreviewRunner injector should not be cached if its getting cached.